wandb/integration/keras/callbacks/metrics_logger.py

```diff
@@ -46,34 +66,64 @@ def __init__(
             tel.feature.keras_metrics_logger = True
 
         if log_freq == "batch":
-            self.log_freq: Union[LogStrategy, int] = 1
-        else:
-            self.log_freq = log_freq
+            log_freq = 1
+
+        self.logging_batch_wise = isinstance(log_freq, int)
+        self.log_freq: Any = log_freq if self.logging_batch_wise else None
         self.global_batch = 0
+        self.global_step = initial_global_step
 
-        if isinstance(log_freq, int):
+        if self.logging_batch_wise:
             # define custom x-axis for batch logging.
             wandb.define_metric("batch/batch_step")
             # set all batch metrics to be logged against batch_step.
             wandb.define_metric("batch/*", step_metric="batch/batch_step")
+        else:
+            # define custom x-axis for epoch-wise logging.
+            wandb.define_metric("epoch/epoch")
+            # set all epoch-wise metrics to be logged against epoch.
+            wandb.define_metric("epoch/*", step_metric="epoch/epoch")
+
+    def _get_lr(self) -> Union[float, None]:
+        if isinstance(self.model.optimizer.learning_rate, tf.Variable):
+            return float(self.model.optimizer.learning_rate.numpy().item())
+        try:
+            return float(
+                self.model.optimizer.learning_rate(step=self.global_step).numpy().item()
+            )
+        except Exception:
+            wandb.termerror("Unable to log learning rate.", repeat=False)
+            return None
```

Comment:
And here.

Reply:
The except block is something we put in there for fail-safe. We don't know of a case where this except case should be trigged. But since logging LR is a relatively minor feature, we don't want our callback erroring out due to any future change in TF.